engine/runtime/src/main/java/org/enso/interpreter/runtime/data/hash/EnsoHashMap.java

```diff
@@ -165,14 +165,15 @@ Type getMetaObject(@CachedLibrary("this") InteropLibrary thisLib) {
 
   @ExportMessage
   @TruffleBoundary
-  Object toDisplayString(boolean allowSideEffects) {
+  Object toDisplayString(
+      boolean allowSideEffects, @CachedLibrary(limit = "5") InteropLibrary interop) {
     var sb = new StringBuilder();
     sb.append("{");
     boolean empty = true;
     for (StorageEntry entry : mapBuilder.getStorage().getValues()) {
       if (isEntryInThisMap(entry)) {
         empty = false;
-        sb.append(entry.key()).append("=").append(entry.value()).append(", ");
+        sb.append(entryToString(entry, interop)).append(", ");
```

Comment:
This ensures that calling `map.to_text` will invoke `to_text` methods on keys and values, if necessary.